wp-cli/vip-two-factor.php

```diff
@@ -1,35 +1,130 @@
+			WP_CLI::error( 'Cannot filter by Two Factor provider when searching for users without Two Factor.' );
+		}
+
+
+		$providers            = Two_Factor_Core::get_providers();
+		$default_provider_map = [
+			'Two_Factor_Email'        => 'email',
+			'Two_Factor_Totp'         => 'totp',
+			'Two_Factor_FIDO_U2F'     => 'fido_u2f',
+			'Two_Factor_Backup_Codes' => 'backup_codes',
+			'Two_Factor_Dummy'        => 'dummy',
+		];
+		$providers_map        = array_intersect_key( $default_provider_map, $providers );
+		if ( $twofa_provider_flag ) {
+			if ( ! in_array( $twofa_provider_flag, $providers_map, true ) ) {
+				WP_CLI::error(
+					sprintf(
+						'Invalid Two Factor provider "%s". Valid values are: %s',
+						$twofa_provider_flag,
+						implode( ', ', array_values( $providers_map ) )
+					)
+				);
+			}
 
-		$users     = get_users();
-		$providers = Two_Factor_Core::get_providers();
+			$twofa_enabled_flag  = 'true'; // In case `--2fa-enabled` was not passed in.
+			$twofa_provider_flag = array_search( $twofa_provider_flag, $providers_map, true );
+		}
 
 		$cap = apply_filters( 'wpcom_vip_two_factor_enforcement_cap', 'manage_options' );
-		foreach ( $users as $user ) {
+
+		if ( $user_id_flag ) {
+			if ( is_numeric( $user_id_flag ) ) {
+				$user = get_user_by( 'id', $user_id_flag );
+			} else {
+				$user = get_user_by( 'login', $user_id_flag );
+				if ( ! $user ) {
+					$user = get_user_by( 'email', $user_id_flag );
+				}
+			}
+			if ( ! $user ) {
+				WP_CLI::error( sprintf( 'User "%s" not found.', $user_id_flag ) );
+			}
+
+			$users = [ $user ];
+		} else {
+			$user_args = [];
+			if ( $twofa_enabled_flag ) {
+				$user_args['capability__in'] = $cap;
+			}
+			if ( $role_flag ) {
+				$user_args['role'] = $role_flag;
+			}
+
+			$users = get_users( $user_args );
+		}
+
+		foreach ( $users as $idx => $user ) {
 			$user->two_factor_enabled   = 'false';
 			$user->two_factor_providers = '';
 
 			if ( Two_Factor_Core::is_user_using_two_factor( $user->ID ) ) {
 				$user->two_factor_enabled = 'true';
 
-				$user_providers             = Two_Factor_Core::get_enabled_providers_for_user( $user );
+				$user_providers = Two_Factor_Core::get_enabled_providers_for_user( $user );
+				if ( $twofa_provider_flag && ! in_array( $twofa_provider_flag, $user_providers, true ) ) {
+					unset( $users[ $idx ] );
+				}
 				$user_providers             = array_map( function ( $provider ) use ( $providers ) {
 					return $providers[ $provider ]->get_label();
 				}, $user_providers );
 				$user->two_factor_providers = implode( ', ', $user_providers );
 			} elseif ( ! user_can( $user->ID, $cap ) ) {
 				$user->two_factor_enabled = 'n/a';
 			}
+
+			if ( $twofa_enabled_flag && $user->two_factor_enabled !== $twofa_enabled_flag ) {
+				unset( $users[ $idx ] );
+			}
 		}
 
+		$fields   = array( 'ID', 'display_name', 'roles' );
 		$fields[] = 'two_factor_enabled';
 		$fields[] = 'two_factor_providers';
```

Comment:
```suggestion
		$fields   = array( 'ID', 'display_name', 'roles', 'two_factor_enabled', 'two_factor_providers' );
```

Further, looking at the output from ` wp user list --format=table`:

```sh
+----+------------------+-------------------+-----------------------------------------+---------------------+---------------+
| ID | user_login       | display_name      | user_email                              | user_registered     | roles         |
+----+------------------+-------------------+-----------------------------------------+---------------------+---------------+
```

...then maybe this report output should also include the user_login, after the ID? 
(May be outside the scope of this PR)

Reply:
Nah, I don't think it's necessary given the user lookup can be login/id/email.